### PR TITLE
refactor(front-matter): remove `createExtractor()`

### PR DESCRIPTION
**Pull request by timreichen**
- removes `createExtractor()`as it seems like overengineering and unnecessarily wrapping functions calls
- renames `_extract()` to `extractAndParse()` for more clarity
- adds `getParserForFormat()` helper function

### CODE REVIEW

File: front_matter/yaml.ts [hunk 1]
```diff
@@ -1,15 +1,12 @@
 // Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
 
-import { createExtractor, type Parser } from "./_create_extractor.ts";
-import { parse } from "@std/yaml/parse";
+import { extractAndParse, type Parser } from "./_shared.ts";
+import { parse as parseYaml } from "@std/yaml/parse";
```
Comment (iuioiua) [inline front_matter/yaml.ts:4]:
```suggestion
import { parse } from "@std/yaml/parse";
```

File: front_matter/toml.ts [hunk 2]
```diff
@@ -1,15 +1,12 @@
 // Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
 
-import { createExtractor, type Parser } from "./_create_extractor.ts";
-import { parse } from "@std/toml/parse";
+import { extractAndParse, type Parser } from "./_shared.ts";
+import { parse as parseToml } from "@std/toml/parse";
```
Comment (iuioiua) [inline front_matter/toml.ts:4]:
```suggestion
import { parse } from "@std/toml/parse";
```

**Review by iuioiua — APPROVED** [review]
This seems more straightforward. Thanks. Could you please rebase?

**Comment by iuioiua** [comment]
PTAL @kt3k

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**